### PR TITLE
Broad share spec

### PR DESCRIPTION
**Pull request by emattei**
SHARE-seq seqspec file for the fastqs that will be submitted by Broad.

### CODE REVIEW

**Review by sbooeshaghi — CHANGES_REQUESTED** [review]
Thank you for taking the time submit a `seqspec` for the `SHARE-Seq` assay! I've added comments for my review. The `spec` has a good structure and a bit more work can make it entirely descriptive of the libraries generated with the SHARE-Seq assay.

The biggest point that needs addressing is the barcode structure. The `ATAC` region didn't contain any information about the `cell_barcodes`. This can be fixed by adding `Regions` in the `ATAC` modality that describe the barcode structure. Additionally, the barcode structure in the `RNA` modality appears to be missing important information about the combinatorial barcoding scheme that generates the final `cell_barcode`. This can be fixed by adding sub-`Regions` in the `cell_barcode` `Region`.

File: assays/SHARE-seq/broad_share_spec.yaml [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+publication_date: 23 October 2020
+description: Simultaneous high-throughput ATAC and RNA expression in the same single cell
+lib_struct: https://teichlab.github.io/scg_lib_structs/methods_html/SHARE-seq.html
+
+modalities:
+- ATAC
+- RNA
+
+assay_spec:
+    - region_id: ATAC
```
[the remark this answers is not in the document]
Comment (emattei) [reply assays/SHARE-seq/broad_share_spec.yaml:14]: For `ATAC` fastq1 and fast2 only contain gDNA. The cell barcode is in the read name. That's why I asked in the IGVF slack if there was a way to describe that.

File: assays/SHARE-seq/broad_share_spec.yaml [hunk 2]
```diff
@@ -0,0 +1,112 @@
+# Assay region
+!Assay
+name: SHARE-Seq
+doi: 10.1016/j.cell.2020.09.056
```
[the remark this answers is not in the document]
Comment (emattei) [reply assays/SHARE-seq/broad_share_spec.yaml:4]: Sure, I will change this.

File: assays/SHARE-seq/broad_share_spec.yaml [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+assay_spec:
+    - region_id: ATAC
+      regions:
+        # Read 1 Fastq
+        - !Region
+          region_id: R1.fastq.gz
+          region_type: R1.fastq.gz
+          name: Read 1 FASTQ
+          sequence_type: random
```
[the remark this answers is not in the document]
Comment (emattei) [reply assays/SHARE-seq/broad_share_spec.yaml:21]: Can I just remove it and put it all in the parent `Region`?

File: assays/SHARE-seq/broad_share_spec.yaml [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+              sequence: X
+              min_len: 50
+              max_len: 50
+              onlist: null
+        # Read 2 Fastq
+        - !Region
+          region_id: R2.fastq.gz
+          region_type: R2.fastq.gz
+          name: Read 2 FASTQ
+          sequence_type: random
```
[the remark this answers is not in the document]
Comment (emattei) [reply assays/SHARE-seq/broad_share_spec.yaml:41]: I guess same thing here.  It seems that I don't have to have sub-`Regions`

File: assays/SHARE-seq/broad_share_spec.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (emattei) [reply assays/SHARE-seq/broad_share_spec.yaml:14]: Oops sorry I missed that. Will fix

File: assays/SHARE-seq/broad_share_spec.yaml [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+              max_len: 50
+              onlist: null
+    - region_id: RNA
+      regions:
+        # Read 1 Fastq
+        - !Region
+          region_id: R1.fastq.gz
+          region_type: R1.fastq.gz
+          name: Read 1 FASTQ
+          sequence_type: random
```
[the remark this answers is not in the document]
Comment (emattei) [reply assays/SHARE-seq/broad_share_spec.yaml:63]: As before.

File: assays/SHARE-seq/broad_share_spec.yaml [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+          name: Read 2 FASTQ
+          sequence_type: joined
+          sequence:
+          min_len: 34
+          max_len: 34
+          onlist: null
+          regions:
+          # Cell Barcode
+            - !Region
+              region_id: cell_barcode
```
[the remark this answers is not in the document]
Comment (emattei) [reply assays/SHARE-seq/broad_share_spec.yaml:91]: We are not going to submit data straight out of the sequencer. What you can see in the synapse is the exact format that we will submit to IGVF and what I am trying to describe in this YAML.

[the remark this answers is not in the document]
Comment (emattei) [reply assays/SHARE-seq/broad_share_spec.yaml:91]: Same as before.

File: assays/SHARE-seq/broad_share_spec.yaml [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+              region_type: cell_barcode
+              name: Barcode
+              sequence_type: onlist
+              sequence: NNNNNNNNNNNNNNNNNNNNNNNN
+              min_len: 24
+              max_len: 24
+              order: 1
+              onlist: !Onlist
+                filename: "gs://broad-buenrostro-bcl-outputs/192_PLATE1_PLATE2_whitelist.txt"
```
[the remark this answers is not in the document]
Comment (emattei) [reply assays/SHARE-seq/broad_share_spec.yaml:100]: Sounds good.

File: assays/SHARE-seq/broad_share_spec.yaml [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+modalities:
+- ATAC
+- RNA
+
+assay_spec:
+    - region_id: ATAC
+      regions:
+        # Read 1 Fastq
+        - !Region
+          region_id: R1.fastq.gz
```
[the remark this answers is not in the document]
Comment (emattei) [reply assays/SHARE-seq/broad_share_spec.yaml:18]: Will fix.

File: assays/SHARE-seq/broad_share_spec.yaml [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+              region_type: gDNA
+              name: Genomic DNA
+              sequence_type: random
+              sequence: X
+              min_len: 50
+              max_len: 50
+              onlist: null
+        # Read 2 Fastq
+        - !Region
+          region_id: R2.fastq.gz
```
[the remark this answers is not in the document]
Comment (emattei) [reply assays/SHARE-seq/broad_share_spec.yaml:38]: As above.

**Comment by emattei** [comment]
* Added the barcode list
* Fixed doi url
* `region_id` is now unique
* Fixed `region_type` is now unique
* `!Region` now present
*  Fixed `sequence_type`
*  Changed barcode definitions

**Comment by sbooeshaghi** [comment]
Hi Eugenio, thank you for making these changes. Could you please install the most recent version of seqspec and run `seqspec check` to verify the correctness and completeness of the spec? In addition, please supply FASTQ files (with 1 million reads) associated with the regions identified in the spec, in your PR. For more information please see the contributing page for more details (https://github.com/IGVF/seqspec/blob/main/docs/CONTRIBUTING.md). Thank you!

**Comment by sbooeshaghi** [comment]
Hey Eugenio- just checking in. I'd like to get your spec merged into the repo!

**Comment by emattei** [comment]
Hi Sina,
Since seqspec is going to be released per file I cannot provide a general version to put here anymore. What you have on the github is already enough as "general" seqspec for SHARE-seq. All the seqspecs relative to the files for IGVF will be in the portal. I guess we can close this pull request.

**Pull request closed without merging**